src-script/zap-combine-reports.js

```diff
@@ -18,50 +18,44 @@
 
 const scriptUtil = require('./script-util.js')
 const fs = require('fs')
+const fsExtra = require('fs-extra')
 
 //workaround: executeCmd()/spawn() fails silently without complaining about missing path to electron
 process.env.PATH = process.env.PATH + ':/usr/local/bin/'
 
-scriptUtil
-  .executeCmd({}, 'mkdir', ['-p', 'reports'])
-  .then(() => {
-    if (fs.existsSync('cypress-coverage/coverage-final.json'))
-      scriptUtil.executeCmd(
-        {},
-        'cp',
-        'cypress-coverage/coverage-final.json reports/from-cypress.json'.split(
-          ' '
-        )
-      )
-  })
 
-  .then(() => {
-    if (fs.existsSync('jest-coverage/coverage-final.json'))
-      scriptUtil.executeCmd({}, 'cp', [
-        'jest-coverage/coverage-final.json',
-        'reports/from-jest.json',
-      ])
-  })
+async function executeScript() {
+  try {
+    // Create directory if it does not exist
+    await fsExtra.ensureDir('reports')
+
+    if (fsExtra.existsSync('cypress-coverage/coverage-final.json')) {
+      await fsExtra.copy('cypress-coverage/coverage-final.json', 'reports/from-cypress.json')
+    }
+
+    if (fsExtra.existsSync('jest-coverage/coverage-final.json')) {
+      await fsExtra.copy('jest-coverage/coverage-final.json', 'reports/from-jest.json')
+    }
+
+    scriptUtil.executeCmd({}, 'npx', ['nyc', 'merge', 'reports'])
+
+    await fsExtra.move('coverage.json', '.nyc_output/out.json', { overwrite: true });
 
-  .then(() => scriptUtil.executeCmd({}, 'npx', ['nyc', 'merge', 'reports']))
-  .then(() =>
-    scriptUtil.executeCmd({}, 'mv', ['coverage.json', '.nyc_output/out.json'])
-  )
-  .then(() =>
     scriptUtil.executeCmd(
       {},
       'npx',
       'nyc report --reporter lcov --reporter text --report-dir coverage'.split(
         ' '
       )
     )
-  )
-  .then(() =>
+
     console.log(
       `✅ Please find the combined report (Jest & Cypress) at ./coverage/lcov-report/index.html`
     )
-  )
 
-  .catch((err) => {
+  } catch (err) {
     console.log(err)
```

Comment:
Good to include file-name#function-name in console message for easy tracking of the issue.
console.log("Error in reporting at zap-combine-reports.js file and executeScript function: " + err)